internal/pkg/api/handleEnroll.go

```diff
@@ -161,7 +161,7 @@ func (et *EnrollerT) fetchStaticTokenPolicy(ctx context.Context, zlog zerolog.Lo
 
 	}
 	// no error, just not found
-	return nil, ErrPolicyNotFound
+	return nil, nil
```

Comment:
Nit. Instead of returning `nil`, `nil`, wdyt about handing `ErrPolicyNotFound` in the caller? It is usually assumed that when a method returns _something_ and an error, this _something_ can be used if the error is nil.

If we decide to keep returning a `nil` error here, maybe we should do the same in [fetchPolicy](https://github.com/elastic/fleet-server/pull/2880/files#diff-b3d2d3a49ef214fba08e3aa10772162cbb65c378e24ef6c4a0c5df65b051f71aR180) to keep the same semantics in both places.

Reply:
The method already return `nil,nil` in some case, it seems a little different than `fetchPolicy` as that method conditionally fetch or not a static token, maybe the method could be renamed to something more clear

Reply:
I renamed the function in https://github.com/elastic/fleet-server/pull/2880/commits/485720e76f2e9f2637f7ecce2ae4c8a4d16b1df2 let me know if it make better sense to you